src/components/ChatEntry.jsx

```diff
@@ -1,20 +1,73 @@
 import './ChatEntry.css';
+import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.jsx';
+
+
+const ChatEntry = (props) => {
+  const likeButtonClicked = () => {
+    props.onLikeToggle(props.id);
+  };
+
+
+  const getSenderType = () => {
+    if (!props.twoPeople) {
+      return '';
+    } else if (props.twoPeople.length === 2) {
+      if (props.sender === props.twoPeople[0]) {
+        return 'chat-entry local';
+      } else if (props.sender === props.twoPeople[1]) {
+        return 'chat-entry remote';
+      }
+    } else if (props.twoPeople.length === 1) {
+      if (props.sender === props.twoPeople[0]) {
+        return 'chat-entry local';
+      }
+    }
+    return '';
+  };
+
+
+  const setColor = () => {
+    if (!props.colorChoice || !props.twoPeople) return '';
+
+    if (props.twoPeople.length === 2) {
+      if (props.sender === props.twoPeople[0]) {
+        return props.colorChoice.vColor;
+      } else if (props.sender === props.twoPeople[1]) {
+        return props.colorChoice.eColor;
+      }
+    } else if (props.twoPeople.length === 1) {
+      return props.colorChoice.vColor;
+    }
+
+    return '';
+  };
+
 
-const ChatEntry = () => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={getSenderType()}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p className = {setColor()}>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp}></TimeStamp></p>
+        <button className="like" onClick={likeButtonClicked}>{props.liked ? '❤️' : '🤍'}</button>
```

Comment:
I suggest dropping these over a couple lines to make it easier for readers to see the component structure at a glance:
```js
<p className="entry-time">
    <TimeStamp time={props.timeStamp} /> // We can use self closing tags for React components
</p>
<button className="like" onClick={likeButtonClicked}>
    {props.liked ? '❤️' : '🤍'}
</button>
```